components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1405,7 +1405,7 @@ private UsersGetResponse filterUsersBySingleAttribute(ExpressionNode node, Map<S
         if (isFilteringNotSupported(node.getOperation())) {
             String errorMessage =
                     "Filter operation: " + node.getOperation() + " is not supported for filtering in users endpoint.";
-            throw new CharonException(errorMessage);
+            throw new BadRequestException(errorMessage);
```

Comment:
Here you can pass the scimType variable as well

As per the attached screenshot, the value in scimType should go to details
scimType should be one of https://datatracker.ietf.org/doc/html/rfc7644#:~:text=A%20SCIM%20detail%20error%20keyword.%20%20See%20Table%209.%20%20OPTIONAL.

Reply:
Hope this is the expected result improved by this commit - https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/531/commits/cc79a63e1512b3c0dde99a68966a025a13cd8870
<img width="713" alt="Screenshot 2024-02-10 at 16 21 05" src="https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/assets/35717390/fb60fbd6-1099-4203-b7af-c7f666572737">